plugins/yallah/src/main/java/de/dfki/vsm/xtension/yallah/YallahExecutor.java

```diff
@@ -0,0 +1,301 @@
+                mLogger.failure("Launch mode " + lm.toString() + "not supported yet. YALLAH not launched.");
+                break;
+            case WebPage:
+                // TODO -- run an internal web server that provides the webpages and open them in a browser.
+                mLogger.failure("Launch mode " + lm.toString() + "not supported yet. YALLAH not launched.");
+                break;
+            default:
+                mLogger.warning("Unrecognized launch method " + lm.toString() );
+        }
+
+
+        //
+        mLogger.message("YALLAH plugin launched.");
+    }
+
+    @Override
+    public void unload() {
+        mLogger.message("YALLAH plugin unloading....");
+
+        //
+        // TODO -- 1. Stop the YALLAH application (Not much to do if it is a WebPage)
+
+        // 2. Shutdown the WebSocket server
+        mWebSocketServer.stop();
+        mSocketConnectCtx = null ;
+
+        mLogger.message("YALLAH plugin unloaded.");
+    }
+
+
+    @Override
+    public void execute(AbstractActivity activity) {
+
+        mLogger.message("YALLAH Agent '" + activity.getActor() + "' said: " + activity.getText());
+        mLogger.message("Activity name: " + activity.getName() + ", type: " + activity.getType() + ", features: " + activity.getFeatures());
+
+        if (activity instanceof SpeechActivity) {
+            //
+            // SPEECH activity
+            SpeechActivity sa = (SpeechActivity) activity;
+
+            String punct = sa.getPunct() ;
+            String text_only = sa.getTextOnly(MARKER).trim() ;
+            LinkedList<String> time_marks = sa.getTimeMarks(MARKER);
+
+            mLogger.message("This is a Speech Activity. text only: '" + text_only + "'; punct: '" + punct + "'"
+                    + "There are " + time_marks.size() + " time marks") ;
+            time_marks.forEach(mLogger::message);
+
+            if (text_only.isEmpty()) {
+                //
+                // If text is empty, there is no need to send the marker to the client:
+                // execute them all of them immediately.
+
+                LinkedList<String> timemarks = sa.getTimeMarks(YallahExecutor.MARKER);
+                for (String tm : timemarks) {
+                    // mLogger.warning("Directly executing activity at timemark " + tm);
+                    mProject.getRunTimePlayer().getActivityScheduler().handle(tm);
+                }
+
+            } else {
+
+                //
+                // Send the sentence to the client
+                String text = activity.getText();
+
+                mSentenceCounter++ ;
+
+                String json_string = "{\n"
+                        + "\"type\": \"text\",\n"
+                        + "\"text\": \"" + text + "\",\n"
+                        + "\"id\": \"" + mSentenceCounter + "\"\n"
+                        + "}";
+
+                // Make text activity blocking
+                //activity.setType(AbstractActivity.Type.blocking);
```

Comment:
Commented out code. Should it not be blocking?

Reply:
Yes, but there is no need to set this property, as the only other place implementing the block is a few lines after: `mActivityWorkerMap.wait();`, and I always need to wait.